### ISSUE TITLE
Do not swallow javascript errors

### ISSUE THREAD

**Issue by Dema** [marmelab/react-admin#3428]
I'm creating a custom dataProvider and having hard time debugging it because all errors are swallowed by notifications, so I can't see full stacktrace.

**What you were expecting:**

When some JS error occurs inside my code I want to be able to see a full error with stack trace in the browser console. At least we could try to do this when `NODE_ENV !== "production"`

**What happened instead:**

Browser console displays translation error for message 
```
Warning: Missing translation for key: "Cannot read property 'data' of null"
```

**Steps to reproduce:**

CodeSandbox (https://codesandbox.io/s/simple-sbzrh)

Open the sandbox. It shows `Warning: Missing translation for key: "AA"` in the browser console instead of full stack trace. Error is thrown in dataProvider like this

```
const dataProvider = () => {
  throw new Error("AA");
};
```

**Other information:**

**Environment**

* React-admin version: 2.9.4

**Comment by Kunnu01** [comment]
@Kmaschta , as you have proposed to use `useTranslate` hook, does this issue needs to be fixed for now using the same approach? I'm willing to work on this.

**Comment by jakewtaylor** [comment]
I'd be happy to give this a go if someone could give me a pointer on how this would work. 😄

**Comment by WiXSL** [comment]
With the new `polyglotI18nProvider` factory you can pass a third parameter to Polyglot with options including how is going to behave regarding missing translations.


```js
const i18nProvider = polyglotI18nProvider(locale => {
    if (locale === 'en') {
        return englishMessages;
    }
    if (locale === 'fr') {
        return import('../i18n/fr.js').then(messages => messages.default);
    }
}, 'en', {warn: () => null});` // <-- Polyglot will use this warn function instead.

**Comment by fzaninotto** [comment]
Fixed by #3927
